docker_101/docker_101.md

```diff
@@ -0,0 +1,427 @@
+
+**So what do you need to keep in mind as you think about how to use Docker in your research?** 
+
+One big difference is the value placed on **transparency** and **interpretability** in the research community. 
+A good, reproducible container for a research project should be not only functional, but easy for others to understand. 
+Because of that, you should think of your Dockerfile as not just a set of instructions but an important form of communication.
+
+This article provides concrete guidance on how to set up your containers for use in research. 
+
+<div class = "external-resource">
+<b style="color: rgb(var(--color-highlight));">External Content</b><br>
+
+Next, read [Ten simple rules for writing Dockerfiles for reproducible data science](https://doi.org/10.1371/journal.pcbi.1008316) by Nüst et al. (2020). 
+
+<br>
+<b style="color: rgb(var(--color-highlight));">Read the whole article, then return here to finish this module.</b>
+</div>
+
+### Quiz: Docker for Research
+
+True or False: The authors recommend including your dataset in the container, to make your analysis fully reproducible.
+
+[( )] True
+[(X)] False
+***
+<div class = "answer">
+
+They recommend storing datafiles outside of the container and using bind mounts to attach the data just when you run the container, rather than building it into the image. 
+See [Rule 7: Mount datasets at run time](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008316#sec020). 
+
+For more guidance on using bind mounts, see the [Docker documentation on bind mounts](https://docs.docker.com/storage/bind-mounts/).
+
+</div>
+***
+
+Which of the following would the authors say is more important?
+
+[(X)] Making your Dockerfile as explicit as possible, so the layers in the image are easier to see
+[( )] Making your Dockerfile as efficient as posisble, so the image is smaller and more portable
+***
+<div class = "answer">
+
+See [Rule 3: Format for clarity](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008316#sec009). 
+In particular:
+
+> In general, if your container is mostly software dependencies, you should not need to worry about image size because (a) your data is likely to have much larger storage requirements; and (b) transparency and inspectability outweigh storage concerns in data science. 
+
+</div>
+***
+
+What character is used to start a comment line in a Dockerfile?
+
+[[#]]
+<script>
+  let input = "@input".trim().toLowerCase();
+  input == "#";
+</script>
+***
+<div class = "answer">
+
+See [Rule 4: Document within the Dockerfile](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008316#sec010) for examples.
+
+</div>
+***
+
+True or False: For a container with complex anlaysis code, the authors recommend writing your Dockerfile so that the whole analysis workflow runs automatically when you start the container.
```

Comment:
```suggestion
True or False: For a container with complex analysis code, the authors recommend writing your Dockerfile so that the whole analysis workflow runs automatically when you start the container.
```